tests/multidevice/multidevice.cpp

```diff
@@ -157,6 +175,258 @@ TEST_F(multidevice_ft, stream_two_2_then_1)
+{
+    k4a_device_t master, subordinate;
+    k4a_fps_t frame_rate = K4A_FRAMES_PER_SECOND_30;
+    uint64_t fps_in_usec = 1000000 / k4a_convert_fps_to_uint(frame_rate);
+
+    ASSERT_EQ(open_master_and_subordinate(&master, &subordinate), K4A_RESULT_SUCCEEDED);
+
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, set_power_and_exposure(master, 8330, 2));
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, set_power_and_exposure(subordinate, 8330, 2));
+
+    k4a_device_configuration_t default_config = K4A_DEVICE_CONFIG_INIT_DISABLE_ALL;
+    default_config.color_format = K4A_IMAGE_FORMAT_COLOR_MJPG;
+    default_config.color_resolution = K4A_COLOR_RESOLUTION_2160P;
+    default_config.depth_mode = K4A_DEPTH_MODE_NFOV_2X2BINNED;
+    default_config.camera_fps = frame_rate;
+    default_config.subordinate_delay_off_master_usec = 0;
+    default_config.depth_delay_off_color_usec = 0;
+    default_config.synchronized_images_only = true;
+
+    k4a_device_configuration_t s_config = default_config;
+    s_config.wired_sync_mode = K4A_WIRED_SYNC_MODE_SUBORDINATE;
+    s_config.depth_delay_off_color_usec = (int32_t)(2 * fps_in_usec * ((uint64_t)rand()) / RAND_MAX - fps_in_usec);
+    s_config.subordinate_delay_off_master_usec = (int32_t)(1 * fps_in_usec * ((uint64_t)rand()) / RAND_MAX);
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_start_cameras(subordinate, &s_config));
+
+    k4a_device_configuration_t m_config = default_config;
+    m_config.wired_sync_mode = K4A_WIRED_SYNC_MODE_MASTER;
+    m_config.depth_delay_off_color_usec = (int32_t)(2 * fps_in_usec * ((uint64_t)rand()) / RAND_MAX - fps_in_usec);
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_start_cameras(master, &m_config));
+
+    printf("Test Running with the following settings:\n");
+    printf("      Master depth_delay_off_color_usec: %d\n", m_config.depth_delay_off_color_usec);
+    printf("         Sub depth_delay_off_color_usec: %d\n", s_config.depth_delay_off_color_usec);
+    printf("  Sub subordinate_delay_off_master_usec: %d\n", s_config.subordinate_delay_off_master_usec);
+
+    printf("\nDelta = Time off master color.\n");
+    printf("Master Color, Master IR(Delta), Sub Color(Delta), Sub IR(Delta)\n");
+    printf("---------------------------------------------------------------\n");
+    for (int x = 0; x < 100; x++)
+    {
+        k4a_capture_t cap_m, cap_s;
+        int64_t ts_m_c, ts_m_ir, ts_s_c, ts_s_ir;
+        k4a_image_t image_c_m, image_ir_m, image_c_s, image_ir_s;
+
+        ASSERT_EQ(K4A_RESULT_SUCCEEDED,
+                  get_syncd_captures(master, subordinate, &cap_m, &cap_s, s_config.subordinate_delay_off_master_usec));
+
+        ASSERT_NE(NULL, (int64_t)(image_c_m = k4a_capture_get_color_image(cap_m)));
+        ASSERT_NE(NULL, (int64_t)(image_c_s = k4a_capture_get_color_image(cap_s)));
+        ASSERT_NE(NULL, (int64_t)(image_ir_m = k4a_capture_get_ir_image(cap_m)));
+        ASSERT_NE(NULL, (int64_t)(image_ir_s = k4a_capture_get_ir_image(cap_s)));
```

Comment:
Why do we need to cast type `k4a_image_t` to a `int64_t` here? Shouldn't this work without the cast?

Reply:
ASSERT_NE has some template logic to it. It doesn't like comparing NULL (int) to k4a_image_t